assets/js/partials/mode-switcher.js

```diff
@@ -0,0 +1,56 @@
+//@preserve Mode Switcher by derekkedziora
+//@preserve License Creative Commons Zero
+
+let systemInitiatedDark = window.matchMedia("(prefers-color-scheme: dark)"); 
+let theme = sessionStorage.getItem('theme');
+
+if (systemInitiatedDark.matches) {
+	document.getElementById("theme-toggle").innerHTML = '<i class="fas fa-adjust fa-rotate-180" aria-hidden="true"></i>';
+} else {
+	document.getElementById("theme-toggle").innerHTML = '<i class="fas fa-adjust" aria-hidden="true"></i>';
+}
+
+function prefersColorTest(systemInitiatedDark) {
+  if (systemInitiatedDark.matches) {
+  	document.documentElement.setAttribute('data-theme', 'dark');		
+   	document.getElementById("theme-toggle").innerHTML = '<i class="fas fa-adjust fa-rotate-180" aria-hidden="true"></i>';
+   	sessionStorage.setItem('theme', '');
+  } else {
+  	document.documentElement.setAttribute('data-theme', 'light');
+    document.getElementById("theme-toggle").innerHTML = '<i class="fas fa-adjust" aria-hidden="true"></i>';
+    sessionStorage.setItem('theme', '');
+  }
+}
+systemInitiatedDark.addListener(prefersColorTest);
+
+
+function modeSwitcher() {
```

Comment:
That's a lot of if else.

Shouldn't it be something like simpler like:
- If dark theme -> switch to light theme
- If light theme -> switch to dark theme

Reply:
It can be ignored if we take the `min.js` from the external "vendor" directly I guess

Reply:
> That's a lot of if else.
> 
> Shouldn't it be something like simpler like:
> 
> * If dark theme -> switch to light theme
> * If light theme -> switch to dark theme

It probably can be refactored but it is not as simple as that. It is more like this:

- default Light
- browserDarkModeInitiated
- userDarkOverride
- userLightOverride